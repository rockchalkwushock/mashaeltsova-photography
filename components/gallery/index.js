import GalleryGrid from './GalleryGrid'

export { GalleryGrid }
