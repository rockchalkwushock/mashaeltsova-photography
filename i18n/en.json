{
  "buttonBooking": "Book Now!",
  "buttonGallery": "View my work here!",
  "formEmail": "Email",
  "formFirstName": "First Name",
  "formLastName": "Last Name",
  "formMessage": "Short Message",
  "formPhone": "Phone",
  "formErrorEmailPattern": "Not a valid email.",
  "formErrorNamePattern":
    "Not a valid pattern. May only contain upper and lowercase letters.",
  "formErrorMessagePattern":
    "May only contain upper & lower case letters and numbers.",
  "formErrorPhonePattern":
    "Should be match one of the following: +11231231234, 11231231234, 1231231234,",
  "formErrorRequired": "Please fill this in.",
  "formErrorTooLong": "Max length 140 characters.",
  "formErrorTooShort": "Minimum length 2 characters.",
  "formRequiredMessage": "All fields required.",
  "galleryNav": "Family,Portrait,Travel,Wedding",
  "modalError": "I'm sorry there seems to be a problem, please try back later.",
  "modalLoading": "Your request is being sent now.",
  "modalSuccess": "Your booking inquiry has been sent!",
  "sectionAbout":
    "For many years now I have had a great passion for capturing a moment in time and sharing the beauty of that moment with the world. I enjoy taking pictures for many varying events. When I am not doing photo shoots you will find me preforming English tutoring, or practicing my other passion, surfing.",
  "sectionPlug":
    "Contact me if you want to relive those precious moments for years to come!",
  "sectionServices":
    "I am willing to negotiate other requested services, however the below are what I primarily offer: Baby Portraits,Family Portraits,Self-Portraits,Engagements,Weddings",
  "sideNav": "Home,Gallery"
}
