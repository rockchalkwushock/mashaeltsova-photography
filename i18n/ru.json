{
  "buttonBooking": "Забронировать сейчас!",
  "buttonGallery": "Мои работы можно посмотреть тут!",
  "formEmail": "Электронный адрес",
  "formFirstName": "Имя",
  "formLastName": "Фамилия",
  "formMessage": "Сообщение",
  "formPhone": "Телефон",
  "formErrorEmailPattern": "FIXME",
  "formErrorNamePattern": "FIXME",
  "formErrorMessagePattern":
    "Может содержать заглавные и строчные буквы, цифры, и .!?,;'",
  "formErrorPhonePattern":
    "Должно соответствовать одному из следующих: +1-234-567-8901, +61-234-567-89-01 +46-234 5678901",
  "formErrorRequired": "Пожалуйста, заполните это поле.",
  "formErrorTooLong": "Максимальная длина 140 знаков.",
  "formErrorTooShort": "минимальная длина 2 знака.",
  "formRequiredMessage": "Необходимо заполнить все поля.",
  "galleryNav": "Семья,Портрет,Путешествия,Свадьбы",
  "modalError": "Произошла ошибка, попробуйте еще раз.",
  "modalLoading": "Ваш запрос отправляется.",
  "modalSuccess": "Ваш запрос на фотосессию отправлен!",
  "sectionAbout":
    "Уже много лет я занимаюсь фотографией и мне нравится моя работа. Мне нравится общаться и фотографировать людей. Если я не фотографирую то, путешествую или учу английский язык, или просто наслаждаюсь жизнью.",
  "sectionPlug": "Связитесь со мной , если хотите обсудить съемку!",
  "sectionServices":
    "Открыта к обсуждению съемок, специализируюсь на следующих съемках: Детская и семейная съемка,Портреты,Помолвки,Свадьбы",
  "sideNav": "FIXME"
}
