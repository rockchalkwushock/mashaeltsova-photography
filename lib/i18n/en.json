{
  "description": "Photography Business Website for Masha Eltsova Photography",
  "greeting": "Hello World!",
  "title": "Masha Eltsova Photography"
}