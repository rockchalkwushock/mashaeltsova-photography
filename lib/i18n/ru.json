{
  "book": "Забронируйте сейчас!",
  "description": "Сайт фотографии для Маши Ельцовой",
  "email": "Эл. адрес",
  "emailInvalid": "Неверный адрес электронной почты.",
  "emailRequired": "Требуется электронная почта.",
  "errorResponse":
    "У вас возникла проблема с вашим запросом, пожалуйста, попробуйте позже.",
  "firstName": "Имя",
  "firstNameRequired": "Имя требуется.",
  "firstNameInvalid": "Неверное имя. Только альфа-символы.",
  "firstNameLength": "Имя должно быть не менее 2 символов.",
  "greeting": "Привет мир!",
  "lastName": "Фамилия",
  "lastNameInvalid": "Недопустимое имя. Только альфа-символы.",
  "lastNameLength": "Фамилия должна быть не менее 2 символов.",
  "lastNameRequired": "Фамилия обязательна.",
  "message": "Каковы ваши потребности в бронировании?",
  "phone": "Номер телефона",
  "phoneInvalid":
    "Неправильный номер телефона. Принимает: +1-234-567-8901, +61-234-567-89-01.",
  "phoneRequired": "Требуется номер телефона.",
  "successResponse": "Ваш запрос на бронирование был отправлен!",
  "title": "Маша Ельцова Фотография",
  "messageRequired": "Сообщение требуется.",
  "messageInvalid": "FIXME",
  "messageMaxLength": "Сообщение должно быть не более 140 символов.",
  "messageMinLength": "Сообщение должно содержать не менее 10 символов.",
  "rcws": "Разработка RCWS"
}
