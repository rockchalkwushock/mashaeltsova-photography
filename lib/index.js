import { initGA, logEvent, logException, logPageView } from './analytics'
import { fetchData, sendDataToMicroService } from './api'
import styles from './styles'
import { createUrl, makeProps, translate } from './utils'
import withIntl from './withIntl'

export {
  createUrl,
  fetchData,
  initGA,
  logEvent,
  logException,
  logPageView,
  makeProps,
  sendDataToMicroService,
  styles,
  translate,
  withIntl
}
