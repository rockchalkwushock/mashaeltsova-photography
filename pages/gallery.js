import { Component } from 'react'
import Head from 'next/head'
import PropTypes from 'prop-types'
import NProgress from 'nprogress'
import { loadGetInitialProps } from 'next/dist/lib/utils'

import { A, Grid, Layout, Photo, SubNav, Thumb } from '../components'
import { fetchData, translate, withIntl } from '../lib'

export class Gallery extends Component {
  static propTypes = {
    photos: PropTypes.array.isRequired,
    intl: PropTypes.object.isRequired,
    locale: PropTypes.string.isRequired,
    messages: PropTypes.object.isRequired,
    url: PropTypes.shape({
      pathname: PropTypes.string.isRequired
    })
  }
  static async getInitialProps(ctx) {
    try {
      const props = await loadGetInitialProps(ctx)
      const { ids } = await fetchData('family')
      return { photos: ids, ...props }
    } catch (err) {
      throw err
    }
  }
  state = {
    currentView: 'family',
    error: false,
    image: null,
    images: null,
    message: '',
    modalPhoto: false
  }
  dismissModal = () => {
    this.setState(state => ({
      ...state,
      image: null,
      modalPhoto: false
    }))
  }
  handleOnClick = async e => {
    const target = translate(e.currentTarget.name)
    try {
      NProgress.start()
      const { error, ids } = await fetchData(target)
      if (error) {
        this.setState(state => ({
          ...state,
          currentView: target,
          error: true,
          image: null,
          images: ids,
          message: error,
          modalPhoto: false
        }))
        NProgress.done()
      }
      this.setState(state => ({
        ...state,
        currentView: target,
        image: null,
        images: ids,
        modalPhoto: false
      }))
      NProgress.done()
    } catch (err) {
      throw err
    }
  }
  showPhoto = id => {
    this.setState(state => ({
      ...state,
      image: id,
      modalPhoto: true
    }))
  }
  renderButtons = () =>
    this.props.messages.galleryNav
      .split(',')
      .map(t => (
        <A
          className="galleryButton"
          key={t}
          onClick={this.handleOnClick}
          text={t}
        />
      ))
  renderError = () => <h1>{this.state.message}</h1>
  renderView() {
    const imgs =
      this.state.currentView === 'family'
        ? this.props.photos
        : this.state.images
    if (this.state.modalPhoto) {
      return (
        <Photo
          onDismiss={() => this.dismissModal()}
          publicId={this.state.image}
        />
      )
    }
    return imgs.map(id => (
      <Thumb
        key={id}
        onClick={() => this.showPhoto(id)}
        publicId={id}
        view={this.state.currentView}
      />
    ))
  }
  render() {
    return (
      <Layout messages={this.props.messages} url={this.props.url}>
        <Head>
          <title>Masha Eltsova Photography | Gallery</title>
          {/* SEO */}
          <meta
            name="description"
            content="Masha Eltsova Photography professional gallery and portfolio."
          />
          {/* SCHEMA.ORG FOR GOOGLE */}
          <meta itemProp="name" content="Masha Eltsova Photography | Gallery" />
          <meta
            itemProp="description"
            content="Masha Eltsova Photography professional gallery and portfolio."
          />
          {/* TWITTER */}
          <meta
            name="twitter:title"
            content="Masha Eltsova Photography | Gallery"
          />
          <meta
            name="twitter:description"
            content="Masha Eltsova Photography professional gallery and portfolio."
          />
          <meta
            name="twitter:url"
            content="https://mashaeltsovaphotography.com/gallery"
          />
          {/* FACEBOOK, PINTREST, GOOGLE+ */}
          <meta
            property="og:title"
            content="Masha Eltsova Photography | Gallery"
          />
          <meta
            property="og:description"
            content="Masha Eltsova Photography professional gallery and portfolio."
          />
          <meta
            property="og:site_name"
            content="Masha Eltsova Photography | Gallery"
          />
          <meta
            property="og:url"
            content="https://mashaeltsovaphotography.com/gallery"
          />
        </Head>
        <SubNav>{this.renderButtons()}</SubNav>
        <Grid>{this.state.error ? this.renderError() : this.renderView()}</Grid>
      </Layout>
    )
  }
}

export default withIntl(Gallery)
