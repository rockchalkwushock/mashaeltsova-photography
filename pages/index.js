import PropTypes from 'prop-types'

import {
  A,
  CollagePhoto,
  Layout,
  Li,
  P,
  Plug,
  Section,
  Ul
} from '../components'
import { BookingForm } from '../containers'
import { withIntl } from '../lib'

export const Index = ({ messages, url }) => {
  const [message, services] = messages.sectionServices.split(': ')
  const list = services.split(',').map(s => <Li key={s} text={s} />)
  return (
    <Layout url={url}>
      <Section className="about">
        <P className="about">{messages.sectionAbout}</P>
      </Section>
      <Section className="services">
        <P className="services">{message}:</P>
        <Ul>{list}</Ul>
      </Section>
      <Section className="gallery">
        <CollagePhoto />
        <A
          className="linkToGallery"
          navigate
          text={messages.buttonGallery}
          url="/gallery"
        />
      </Section>
      <Section className="booking">
        <Plug messages={messages} />
        <BookingForm messages={messages} />
      </Section>
    </Layout>
  )
}

Index.propTypes = {
  intl: PropTypes.object.isRequired,
  locale: PropTypes.string.isRequired,
  messages: PropTypes.object.isRequired,
  url: PropTypes.shape({
    pathname: PropTypes.string.isRequired
  })
}

export default withIntl(Index)
