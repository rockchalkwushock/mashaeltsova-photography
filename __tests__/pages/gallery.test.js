import React from 'react'
import { render } from 'enzyme'

import { Gallery } from '../../pages/gallery'
import en from '../../i18n/en.json'
import ru from '../../i18n/ru.json'
import { makeProps } from '../../lib'

const ids = ['a', 'b', 'c']

describe('Component: Gallery', () => {
  test('Renders without exploding in English', () => {
    const tree = render(
      <Gallery {...makeProps('en', en, '/gallery')} photos={ids} />
    )
    expect(tree).toMatchSnapshot()
  })
  test('Renders without exploding in Russian', () => {
    const tree = render(
      <Gallery {...makeProps('ru', ru, '/gallery')} photos={ids} />
    )
    expect(tree).toMatchSnapshot()
  })
})
