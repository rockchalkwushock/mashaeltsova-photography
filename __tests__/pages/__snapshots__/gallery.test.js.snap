// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Gallery Renders without exploding in English 1`] = `
<div
  class="jsx-3955059306 wrapper"
>
  <div
    class="jsx-592563972"
  />
  <header
    class="jsx-1518572121 gallery"
  >
    <picture>
      <source
        media="(min-width: 1024px)"
        srcset="https://res.cloudinary.com/undefined/c_scale,dpr_auto,f_auto,o_90,q_auto:best,w_1024/watermark"
      />
      <source
        media="(min-width: 768px)"
        srcset="https://res.cloudinary.com/undefined/c_scale,dpr_auto,f_auto,o_90,q_auto:best,w_768/watermark"
      />
      <source
        media="(min-width: 540px)"
        srcset="https://res.cloudinary.com/undefined/c_scale,dpr_auto,f_auto,o_90,q_auto:best,w_540/watermark"
      />
      <source
        media="(min-width: 425px)"
        srcset="https://res.cloudinary.com/undefined/c_scale,dpr_auto,f_auto,o_90,q_auto:best,w_425/watermark"
      />
      <img
        alt="Masha Eltsova Photography Logo"
        src="https://res.cloudinary.com/undefined/c_scale,dpr_auto,f_auto,o_90,q_auto:best,w_320/watermark"
      />
    </picture>
  </header>
  <main
    class="jsx-1845556999 gallery"
  >
    <nav
      class="jsx-709412855"
    >
      <button
        class="jsx-3153731447"
        name="family"
      >
        Family
      </button>
      <button
        class="jsx-3153731447"
        name="portrait"
      >
        Portrait
      </button>
      <button
        class="jsx-3153731447"
        name="travel"
      >
        Travel
      </button>
      <button
        class="jsx-3153731447"
        name="wedding"
      >
        Wedding
      </button>
    </nav>
    <div
      class="jsx-4021821574"
    >
      <div
        class="lazyload-placeholder"
        style="height:100px"
      />
      <div
        class="lazyload-placeholder"
        style="height:100px"
      />
      <div
        class="lazyload-placeholder"
        style="height:100px"
      />
    </div>
  </main>
  <footer
    class="jsx-747610455"
  >
    <div
      class="jsx-590062731 social"
    >
      <a
        href="https://www.facebook.com/masha.eltsova"
      >
        <span
          aria-hidden="true"
          class="fa fa-facebook fa-2x"
        />
      </a>
      <a
        href="https://www.instagram.com/mashaeltcovaphotography"
      >
        <span
          aria-hidden="true"
          class="fa fa-instagram fa-2x"
        />
      </a>
      <a
        href="https://www.vk.com/club65938200"
      >
        <span
          aria-hidden="true"
          class="fa fa-vk fa-2x"
        />
      </a>
    </div>
    <div
      class="jsx-590062731 copyright"
    >
      <a
        class="jsx-1299221301"
        href="https://rcws-development.com"
      >
        <span
          class="jsx-1299221301"
        >
          Designed by RCWS Development 2017
        </span>
      </a>
    </div>
  </footer>
</div>
`;

exports[`Component: Gallery Renders without exploding in Russian 1`] = `
<div
  class="jsx-3955059306 wrapper"
>
  <div
    class="jsx-592563972"
  />
  <header
    class="jsx-1518572121 gallery"
  >
    <picture>
      <source
        media="(min-width: 1024px)"
        srcset="https://res.cloudinary.com/undefined/c_scale,dpr_auto,f_auto,o_90,q_auto:best,w_1024/watermark"
      />
      <source
        media="(min-width: 768px)"
        srcset="https://res.cloudinary.com/undefined/c_scale,dpr_auto,f_auto,o_90,q_auto:best,w_768/watermark"
      />
      <source
        media="(min-width: 540px)"
        srcset="https://res.cloudinary.com/undefined/c_scale,dpr_auto,f_auto,o_90,q_auto:best,w_540/watermark"
      />
      <source
        media="(min-width: 425px)"
        srcset="https://res.cloudinary.com/undefined/c_scale,dpr_auto,f_auto,o_90,q_auto:best,w_425/watermark"
      />
      <img
        alt="Masha Eltsova Photography Logo"
        src="https://res.cloudinary.com/undefined/c_scale,dpr_auto,f_auto,o_90,q_auto:best,w_320/watermark"
      />
    </picture>
  </header>
  <main
    class="jsx-1845556999 gallery"
  >
    <nav
      class="jsx-709412855"
    >
      <button
        class="jsx-3153731447"
        name="семья"
      >
        Семья
      </button>
      <button
        class="jsx-3153731447"
        name="портрет"
      >
        Портрет
      </button>
      <button
        class="jsx-3153731447"
        name="путешествия"
      >
        Путешествия
      </button>
      <button
        class="jsx-3153731447"
        name="свадьбы"
      >
        Свадьбы
      </button>
    </nav>
    <div
      class="jsx-4021821574"
    >
      <div
        class="lazyload-placeholder"
        style="height:100px"
      />
      <div
        class="lazyload-placeholder"
        style="height:100px"
      />
      <div
        class="lazyload-placeholder"
        style="height:100px"
      />
    </div>
  </main>
  <footer
    class="jsx-747610455"
  >
    <div
      class="jsx-590062731 social"
    >
      <a
        href="https://www.facebook.com/masha.eltsova"
      >
        <span
          aria-hidden="true"
          class="fa fa-facebook fa-2x"
        />
      </a>
      <a
        href="https://www.instagram.com/mashaeltcovaphotography"
      >
        <span
          aria-hidden="true"
          class="fa fa-instagram fa-2x"
        />
      </a>
      <a
        href="https://www.vk.com/club65938200"
      >
        <span
          aria-hidden="true"
          class="fa fa-vk fa-2x"
        />
      </a>
    </div>
    <div
      class="jsx-590062731 copyright"
    >
      <a
        class="jsx-1299221301"
        href="https://rcws-development.com"
      >
        <span
          class="jsx-1299221301"
        >
          Designed by RCWS Development 2017
        </span>
      </a>
    </div>
  </footer>
</div>
`;
